packages/react-ui/src/TableOfContent/useTOCReducer.js

```diff
@@ -0,0 +1,32 @@
+import { useReducer, useCallback } from "react";
+
+export const rootReducer = (state, action) => {
+  switch (action.type) {
+    case "setTitles":
+      return { titles: [...action.payload.map((title) => ({ ...title }))] };
+    case "setActive":
+      return {
+        titles: [
+          ...state.titles.map((title) => ({
+            ...title,
+            active: title.id === action.payload ? true : false,
+          })),
+        ],
+      };
+  }
+};
+
+export const useTOCReducer = (initialState) => {
+  const [state, dispatch] = useReducer(rootReducer, initialState);
+  const observerCallback = useCallback((entries) => {
```

Comment:
I wonder if we could test that part too calling the callback by hand

Reply:
The issue is that I can't instantiate useTOCReducer outside of a function component so this make it harder to test it. Any idea on how I could test it without complexifying too much the base code ?

Reply:
I thought you could test it like useRootReducer by calling the function. ( and assert on title changes when calling observableCallback or setTitle)

Reply:
Hum not sure how I can test useTOCReducer inside a react function inside a jest test